File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.swift
import Foundation

func solution(_ sizes:[[Int]]) -> Int {

    var (ansMax, ansMin) = (0, 0)
    
    for i in sizes {
        ansMax = max(ansMax, i.max()!)
        print(i)
        ansMin = max(ansMin, i.min()!)
        print(ansMin)
    }
    
    return ansMax * ansMin
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int maxWidth =0;
        int maxHeight =0;
        
        for(int[] size: sizes) {
            int w =  size[0];
            int h = size[1];
            
            int bigger = Math.max(w, h);
            int smaller = Math.min(w, h);
            
            if(bigger > maxWidth) maxWidth = bigger;
            if (smaller > maxHeight) maxHeight = smaller;
        }
        
        return maxWidth * maxHeight ;
    }
}